### PR TITLE
FOGL-7748: Added support for JSONArray type in Reading class

### PR DESCRIPTION
**Pull request by gnandan**
Unit test case result : 
```
[==========] Running 3 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 3 tests from TESTReading
[ RUN      ] TESTReading.TestReadingForNestedListType
[       OK ] TESTReading.TestReadingForNestedListType (1 ms)
[ RUN      ] TESTReading.TestUnspportedReadingForListType
[       OK ] TESTReading.TestUnspportedReadingForListType (3 ms)
[ RUN      ] TESTReading.TestReadingForListType
[       OK ] TESTReading.TestReadingForListType (2 ms)
[----------] 3 tests from TESTReading (8 ms total)

[----------] Global test environment tear-down
[==========] 3 tests from 1 test case ran. (11 ms total)
[  PASSED  ] 3 tests.
```

### CODE REVIEW

File: C/common/reading.cpp [hunk 1]
```diff
@@ -590,6 +604,20 @@ vector<Datapoint *> *values = new vector<Datapoint *>;
...
+			for (auto& v : itr->value.GetArray())
+			{
+				if (v.IsNumber())
+					arr.push_back(v.GetDouble());
+				else
+					throw runtime_error("Only numeric lists are currently supported in datapoints");
+			}
+
+			DatapointValue dpv(arr);
+			values->push_back(new Datapoint(name, dpv));
```
Comment (Innovative-ashwin) [inline C/common/reading.cpp:619]: emplace_back

File: C/common/reading.cpp [hunk 2]
```diff
@@ -590,6 +604,20 @@ vector<Datapoint *> *values = new vector<Datapoint *>;
...
 		}
+		else if (itr->value.IsArray())
+		{
+			vector<double> arr;
+			for (auto& v : itr->value.GetArray())
+			{
+				if (v.IsNumber())
+					arr.push_back(v.GetDouble());
+				else
+					throw runtime_error("Only numeric lists are currently supported in datapoints");
```
Comment (Innovative-ashwin) [inline C/common/reading.cpp:615]: are we catching this exception anywhere?

**Review by MarkRiddoch — COMMENTED** [review]
Looks good, although worth using emplace_back for performance

File: C/common/reading.cpp (hunk 1 above)
Comment (gnandan) [reply C/common/reading.cpp]: fixed

File: C/common/reading.cpp (hunk 2 above)
Comment (gnandan) [reply C/common/reading.cpp:615]: Caller is supposed to handle this exception like other existing runtime_error

**Review by Innovative-ashwin — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**